Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2764,45 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-
+  if FClientInfo.ClientName <> '' then begin
+    if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := 'Name='+FClientInfo.FClientName;
+      if FClientInfo.FClientVersion <> '' then begin
+        LClnt := LClnt + '; Version='+FClientInfo.FClientVersion;
+      end;
+      if FClientInfo.FVendor <> '' then begin
+        LClnt := LClnt + '; Vendor='+FClientInfo.FVendor;
+      end;
+      SendCmd('CSID ' + LClnt);  {do not localize}
+      LBuf := TrimRight(Self.LastCmdResult.Text.Text);

```

Comment:
If the `CSID` command fails, the response should not be parsed.  It may be a non-empty string but will not be in a format that your parser is expecting.